### PR TITLE
[Experimental] Remote cluster

### PR DESCRIPTION
**Pull request by PatrickDuncan**
Just for internal reviewing

The Kiali deployment change will be a Kiali Operator change eventually

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -472,6 +472,55 @@ spec:
...
+
+=== Observing a Remote Cluster
+
+There are certain use cases where Kiali needs to be deployed in one cluster (Control Plane) and observe a different cluster (Data Plane). link:https://user-images.githubusercontent.com/6889074/87819080-ad099980-c839-11ea-834b-56eec038ce4d.png:[Diagram].
+
+Follow these steps:
+
+1: You should have one of the central istiod setups running
+
+  - https://github.com/istio/istio/wiki/Central-Istiod-manages-remote-data-plane
```
Comment (linsun) [inline README.adoc]: I would remove this one

File: README.adoc [hunk 2]
```diff
@@ -472,6 +472,55 @@ spec:
...
+
+  - https://github.com/istio/istio/wiki/Central-Istiod-single-cluster-steps
+
+2: Deploy the link:https://github.com/istio/istio/blob/master/samples/addons/kiali.yaml[Kiali ClusterRole, ClusterRoleBinding, and ServiceAccount] to the Data Plane
```
Comment (linsun) [inline README.adoc]: would not this install kiali as well?

File: README.adoc [hunk 3]
```diff
@@ -472,6 +472,55 @@ spec:
...
+
+3: Create a remote secret in the Control Plane, using the Data Plane ServiceAccount you just created. This allows the Control Plane to read from and modify the Data Plane
+[source,shell]
+----
+istioctl x create-remote-secret --service-account kiali-service-account --context=$DataPlane --name kiali | kubectl apply -n istio-system --context=$ControlPlane -f -
+----
+
+4: Run the Kiali Deployment with the following additions. Kiali will use the API server from the remote secret instead of the local API server.
```
Comment (linsun) [inline README.adoc]: This step isn't clear.  also which cluster?

File: README.adoc (hunk 2 above)
Comment (linsun) [reply README.adoc]: also do you mean data plane cluster here?

Comment (PatrickDuncan) [reply README.adoc]: So this step is just to create the Kiali specific ClusterRole, ClusterRoleBinding, and ServiceAccount to the Data Plane. Not the Kiali Deployment, Service, etc.

File: README.adoc (hunk 3 above)
Comment (PatrickDuncan) [reply README.adoc]: Kiali would be deployed to the Control Plane cluster.

I expect someone to take the existing Kiali Deployment and then add the Volume and VolumeMounts specified. I'll say that explicitly

File: README.adoc [hunk 4]
```diff
@@ -472,6 +472,55 @@ spec:
...
+      - volumeMounts:
+        - mountPath: /kiali-remote-secret
+          name: kiali-remote-secret
+      volumes:
+      - name: kiali-remote-secret
+        secret:
+          defaultMode: 420
+          optional: true
+          secretName: istio-remote-secret-kiali
+----
```
Comment (linsun) [inline README.adoc:509]: we have similar code in istio, we set the mounted path as env var.

File: README.adoc (hunk 2 above)
Comment (linsun) [reply README.adoc]: i see.   basically just portions of the kiali yaml.  You can check if Kiali team allows you to configure it via their operator.

Comment (PatrickDuncan) [reply README.adoc]: I took a quick look at the [spec](https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml) and it doesn't seem like it allows you to turn entire resources on/off like Istio.
We'll need to add 2 options in the future
1. Only include the CR, CRB, and SA
2. Exclude the CR, CRB, and SA and add volumes to the Deployment

Lots to learn :)

**Pull request closed without merging**